pkg/storage/gcs/gcs.go

```diff
@@ -1,53 +1,128 @@
 package gcs
 
 import (
+	"context"
 	"fmt"
 	"reflect"
+	"strconv"
+	"strings"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/rest"
 
+	gstorage "cloud.google.com/go/storage"
+	goauth2 "golang.org/x/oauth2/google"
+	gapi "google.golang.org/api/googleapi"
+	goption "google.golang.org/api/option"
+
+	configapiv1 "github.com/openshift/api/config/v1"
 	operatorapi "github.com/openshift/api/operator/v1"
 	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
 	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
 	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
 )
 
+var (
+	gcsClient *gstorage.Client
+	gcsConfig *GCS
```

Comment:
Please don't use global variables. They persist between `sync`s and may contain outdated information.

Reply:
updated in next push